File: poly.cs
interface Cmd {
    void run();
}

class Cmd_A : Cmd {
    public void run() {
        System.Console.Write("A");
    }
}

class Cmd_B : Cmd {
    public void run() {
        System.Console.Write("B");
    }
}

class poly {
    static void Main() {
        Cmd[] cmds = new Cmd[3];
        cmds[0] = new Cmd_A();
        cmds[1] = new Cmd_B();
        cmds[2] = new Cmd_A();
        int i;
        for (i = 0; i < 3; i++)
            cmds[i].run();
        System.Console.Write("\n");
    }
}

File: poly.c
#include <stdio.h>

typedef void (*cmd_t)();

void cmd_a() {
    printf("A");
}

void cmd_b() {
    printf("B");
}

int main() {
    cmd_t cmds[3];
    cmds[0] = &cmd_a;
    cmds[1] = &cmd_b;
    cmds[2] = &cmd_a;
    int i;
    for (i = 0; i < 3; i++)
        cmds[i]();
    puts("");
}
